cf-execd/cf-execd.c

```diff
@@ -553,20 +568,56 @@ static int SetupRunagentSocket()
     int ret;
     if (RUNAGENT_SOCKET_DIR == NULL)
     {
-        ret = snprintf(sock_info.sun_path, sizeof(sock_info.sun_path) - 1,
+        ret = snprintf(sock_info->sun_path, sizeof(sock_info->sun_path) - 1,
                        "%s/cf-execd.sockets/"CF_EXECD_RUNAGENT_SOCKET_NAME, GetStateDir());
     }
     else
     {
-        ret = snprintf(sock_info.sun_path, sizeof(sock_info.sun_path) - 1,
+        ret = snprintf(sock_info->sun_path, sizeof(sock_info->sun_path) - 1,
                        "%s/"CF_EXECD_RUNAGENT_SOCKET_NAME, RUNAGENT_SOCKET_DIR);
     }
+    return ret;
+}
+
+static inline bool SetRunagentSocketACLs(char *sock_path, StringSet *allow_users)
+{
+    /* Allow access to the socket (rw) */
+    bool success = AllowAccessForUsers(sock_path, allow_users, true, false);
+
+    /* Need to ensure access to the parent folder too (rx) */
+    if (success)
+    {
+        ChopLastNode(sock_path);
+        success = AllowAccessForUsers(sock_path, allow_users, false, true);
+    }
+    return success;
+}
+
+static int SetupRunagentSocket(const ExecdConfig *execd_config)
+{
+    assert(execd_config != NULL);
+
+    int runagent_socket = -1;
+
+    struct sockaddr_un sock_info;
+    int ret = GetRunagentSocketInfo(&sock_info);
```

Comment:
Does it make more sense for the `GetRunagentSocketInfo` function do the `assert(ret > 0)` and truncation testing? Does it not make more sense for this function to return `-1` on error?

It's a bit confusing to me to check for truncation when calling a getter function.

Reply:
Yup, makas sense. I wasn't really thinking of it as a function, `static inline` is still just a C variant of copy-paste to me :grin: